### ISSUE TITLE
Decompiler treats multiple conditions within an if check as separate checks

### ISSUE THREAD

**Issue by NarryG** [krzys-h/UndertaleModTool#17]
Another QOL improvement request for the decompiler.  Currently, if you have a conditional check with multiple parameters such as
```
if(foo = bar && dog == cat)
   DoStuff();
```

it gets treated as
```
if(foo == bar)
  _temp_local_var_1 = (dog == cat)
else 
   _temp_local_var_1 = 0
if(temp_local_var_1)
   DoStuff()
```  
Just for reference at how much more readable this is, here's a comparison between the UndertaleModTool output and the GMDSam output (doesn't work with Bytecode 16).  

UndertaleModTool Output  

```
if (self.con == 12)
{
    1584.cutscene = 1 // 1584 = obj_mainchara (UndertaleGameObject)
    __view_set(9, 0, -4)
    self.siner = (self.siner + 1)
    if (self.alarm[4] > 20)
        _temp_local_var_766 = (self.intense < 1.5) // 1 = obj_musicobjectparent (UndertaleGameObject)
    else
        _temp_local_var_766 = 0
    if _temp_local_var_766
        self.intense = (self.intense + 0.02) // 0 = obj_backgrounder_parent (UndertaleGameObject)
    if (self.alarm[8] > 1)
        _temp_local_var_767 = (self.intense < 0.6) // 0 = obj_backgrounder_parent (UndertaleGameObject)
    else
        _temp_local_var_767 = 0
    if _temp_local_var_767
        self.intense = (self.intense + 0.01) // 0 = obj_backgrounder_parent (UndertaleGameObject)
    else
    {
        if (self.intense > 0)
            self.intense = (self.intense - 0.1) // 0 = obj_backgrounder_parent (UndertaleGameObject)
    }
    __view_set(0, 0, (self.xx + (sin((self.siner / 1.3)) * self.intense))) // 1 = obj_musicobjectparent (UndertaleGameObject)
    __view_set(1, 0, (self.yy + (sin((self.siner / 0.9)) * self.intense))) // 0 = obj_backgrounder_parent (UndertaleGameObject)
}

```

GMDSam output  

```
 if(self.con == 12) {
        obj_mainchara.cutscene= 1;
        self.view_object[0]= -4;
        self.siner++;
        if(self.alarm[4] > 20 && self.intense < 1.5)
             self.intense+= 0.02;
        if(self.alarm[8] > 1 && self.intense < 0.6)
             self.intense+= 0.01;
         else  {
            if(self.intense > 0) self.intense-= 0.1;
        }
        self.view_xview[0]= self.xx + sin(self.siner / 1.3) * self.intense;
        self.view_yview[0]= self.yy + sin(self.siner / 0.9) * self.intense;
    }
```

Would be great if this could be unraveled as single checks using && or || rather than multiple checks.

**Comment by krzys-h** [comment]
This is another thing I want to implement eventually. The reason this happens is that GML does short-circuit evaluation, meaning that if you have an expression like `if (false && something_else())` then something_else() will never be executed.

**Comment by PoroCYon** [comment]
[Relevant paper](https://www.ndss-symposium.org/ndss2015/ndss-2015-programme/no-more-gotos-decompilation-using-pattern-independent-control-flow-structuring-and-semantics/)

**Comment by krzys-h** [comment]
Implemented in #162
